### PR TITLE
Implement preemptible work items for user mode

### LINKED ISSUE #854: Implement preemptible work items for user mode.
### Describe the feature you'd like supported

Currently `ebpf_queue_preemptible_work_item`, `ebpf_allocate_preemptible_work_item` returns `EBPF_OPERATION_NOT_SUPPORTED` for user mode. Implement these APIs for user mode also.

### Proposed solution

Use NT thread pool APIs to implement this.
https://docs.microsoft.com/en-us/windows/win32/api/threadpoolapiset/nf-threadpoolapiset-createthreadpoolwork

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by saxena-anurag**
## Description

Add support for preemptible work items for user mode.

## Testing

Existing tests cover the new code added.

## Documentation

NA

Fixes #854

### CODE REVIEW

**Comment by Alan-Jowett** [comment]
Note:
Codecoverage is marking as failed as only 72% of the change was hit by tests compared to 85% in the baseline.

**Comment by saxena-anurag** [comment]
> Note: Codecoverage is marking as failed as only 72% of the change was hit by tests compared to 85% in the baseline.

This is mainly due to the code added to handle possible threadpool API failures. Those code paths are not covered with our tests.

**Comment by Alan-Jowett** [comment]
> > Note: Codecoverage is marking as failed as only 72% of the change was hit by tests compared to 85% in the baseline.
> 
> This is mainly due to the code added to handle possible threadpool API failures. Those code paths are not covered with our tests.

See:
https://docs.codecov.com/docs/codecovyml-reference

This has options to set the required % covered on a change to show as pass.

File: libs/platform/user/ebpf_platform_user.cpp [hunk 1]
```diff
@@ -20,15 +21,78 @@ bool _ebpf_platform_is_preemptible = true;
 
 extern "C" bool ebpf_fuzzing_enabled = false;
 
+// Thread pool related globals.
+static TP_CALLBACK_ENVIRON _callback_environment;
+static PTP_POOL _pool = NULL;
+static PTP_CLEANUP_GROUP _cleanup_group = NULL;
```
Comment (dthaler) [inline libs/platform/user/ebpf_platform_user.cpp:27]:
```suggestion
static PTP_POOL _pool = nullptr;
static PTP_CLEANUP_GROUP _cleanup_group = nullptr;
```
nit: use C++ keyword nullptr in .cpp files wherever possible (same in line 37, 51, etc.)

**Review by dthaler — DISMISSED** [review]

File: libs/platform/user/ebpf_platform_user.cpp (hunk 1 above)
Comment (saxena-anurag) [reply libs/platform/user/ebpf_platform_user.cpp]: Fixed all usage of `NULL` in this file.

**Review by dthaler — APPROVED** [review]

**Review by Alan-Jowett — APPROVED** [review]

**Pull request merged**